tests/test_testserver.py

```diff
@@ -27,6 +27,12 @@ def handler(sock):
             assert text == answer
             sock.close()
 
+
+
+
+
+
+
```

Comment:
**Suggestion:** Remove unnecessary blank lines. They do not improve readability and can make the code longer than necessary.
```suggestion
def test_server_closes(self):
    """the server closes when leaving the context manager"""
    with Server.basic_response_server() as (host, port):
def test_text_response(self):
    """the text_response_server sends the given text"""
    server = Server.text_response_server(
```